supervisor/hardware/disk.py

```diff
@@ -0,0 +1,120 @@
+"""Read disk hardware info from system."""
+import logging
+from pathlib import Path
+import shutil
+from typing import Union
+
+from ..coresys import CoreSys, CoreSysAttributes
+from .const import UdevSubsystem
+from .data import Device
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+_MOUNTINFO: Path = Path("/proc/self/mountinfo")
+_BLOCK_DEVICE_CLASS = "/sys/class/block/{}"
+_BLOCK_DEVICE_EMMC_LIFE_TIME = "/sys/block/{}/device/life_time"
+
+
+class HwDisk(CoreSysAttributes):
+    """Representation of an interface to disk utils."""
+
+    def __init__(self, coresys: CoreSys):
+        """Init hardware object."""
+        self.coresys = coresys
+
+    def is_system_partition(self, device: Device) -> bool:
+        """Return true if this is a system disk/partition."""
+        if device.subsystem != UdevSubsystem.DISK:
+            return False
+
+        if device.attributes.get("ID_FS_LABEL", "").startswith("hassos"):
+            return True
+        return False
+
+    def get_disk_total_space(self, path: Union[str, Path]) -> float:
+        """Return total space (GiB) on disk for path."""
+        total, _, _ = shutil.disk_usage(path)
```

Comment:
instead of querying this for all 3 (which only use 1 or the returned values), can we do some cache here?

Reply:
right, let handle it on a new PR, I just copy that over